ui/raidboss/data/03-hw/trial/thordan-ex.ts

```diff
@@ -0,0 +1,513 @@
+      run: (data, matches) => {
+        if (data.swordKnight === matches.source)
+          data.swordTarget = matches.target;
+        if (data.shieldKnight === matches.source)
+          data.shieldTarget = matches.target;
+      },
+    },
+    {
+      id: 'ThordanEX Holy Bladedance Call',
+      type: 'StartsUsing',
+      netRegex: { id: '1496', source: ['Ser Adelphel', 'Ser Janlenoux'], capture: false },
+      delaySeconds: 0.5,
+      suppressSeconds: 5,
+      alertText: (data, _matches, output) => {
+        if (data.swordTarget === undefined || data.shieldTarget === undefined)
+          return output.unknownDance!();
+        const swordTarget = data.ShortName(data.swordTarget);
+        const shieldTarget = data.ShortName(data.shieldTarget);
+        if (data.swordTarget === data.shieldTarget)
+          return output.singleDance!({ target: swordTarget });
+        return output.doubleDance!({ sword: swordTarget, shield: shieldTarget });
+      },
+      run: (data) => {
+        delete data.shieldKnight;
+        delete data.swordKnight;
+        delete data.shieldTarget;
+        delete data.swordTarget;
+      },
+      outputStrings: {
+        unknownDance: {
+          en: 'Heavy busters',
+        },
+        singleDance: {
+          en: '2x buster on ${target}',
+        },
+        doubleDance: {
+          en: 'Sword buster on ${sword} (shield on ${shield})',
+        },
+      },
+    },
+    {
+      id: 'ThordanEX Skyward Leap',
+      type: 'HeadMarker',
+      netRegex: { id: '000E' },
+      condition: Conditions.targetIsYou(),
+      alarmText: (data, _matches, output) => {
+        if (data.phase !== 2)
+          return output.defamationNoNumber!();
+        return output.defamationCounted!({ number: data.defCounter });
+      },
+      run: (data) => data.defCounter += 1,
+      outputStrings: {
+        defamationNoNumber: {
+          en: 'Defamation on YOU',
+        },
+        defamationCounted: {
+          en: 'Defamation #${number} on YOU',
+        },
+      },
+    },
+    {
+      id: 'ThordanEX Spiral Pierce Collect',
+      type: 'Tether',
+      netRegex: { id: '0005' },
+      run: (data, matches) => data.pierceTargets.push(matches.target),
+    },
+    {
+      id: 'ThordanEX Spiral Pierce',
+      type: 'Tether',
+      netRegex: { id: '0005', capture: false },
+      delaySeconds: 0.5,
+      suppressSeconds: 5,
+      alarmText: (data, _matches, output) => {
+        if (!data.pierceTargets.includes(data.me))
+          return;
+        return output.pierceYou!();
+      },
+      alertText: (data, _matches, output) => {
+        if (data.pierceTargets.includes(data.me))
+          return;
+        return output.pierceOthers!();
+      },
+      run: (data) => data.pierceTargets = [],
+      outputStrings: {
+        pierceYou: {
+          en: 'Line AoE on YOU',
+        },
+        pierceOthers: {
+          en: 'Avoid tether line AoEs',
```

Comment:
Do you think this should be dropped for noise reasons? Do folks always have ice if they don't have a tether?

Reply:
They do, yes, that's reasonable. The tethers should be visible enough to avoid.